src/main/java/org/testeditor/fixture/web/WebDriverFixture.java

```diff
@@ -288,15 +308,115 @@ private void launchFirefox() {
      */
     private void launchInternetExplorer() {
         setupDrivermanager(InternetExplorerDriverManager.getInstance());
-        DesiredCapabilities capabilities = DesiredCapabilities.internetExplorer();
-        setCapabilities(capabilities);
-        driver = new InternetExplorerDriver(capabilities);
+        String browserName = BrowserType.IE;
+        DesiredCapabilities desiredCapabilities = DesiredCapabilities.internetExplorer();
+        setCapabilitiesForIe(desiredCapabilities, browserName);
+        driver = new InternetExplorerDriver(desiredCapabilities);
         registerShutdownHook(driver);
     }
 
-    private void setCapabilities(DesiredCapabilities capabilities) {
-        capabilities.setCapability(InternetExplorerDriver.INTRODUCE_FLAKINESS_BY_IGNORING_SECURITY_DOMAINS, true);
-        capabilities.setCapability("requireWindowFocus", true);
+    private void setCapabilitiesForIe(DesiredCapabilities desiredCapabilities, String browserName) {
+        List<BrowserSetting> capabilities = new ArrayList<>();
+        List<BrowserSetting> options = new ArrayList<>();
+        populateBrowserSpecificSettings(browserName, capabilities, options);
+        populateCapabilities(desiredCapabilities, capabilities);
+    }
+
+    private Object populateBrowserSettings(DesiredCapabilities desiredCapabilities, String browserName) {
+        List<BrowserSetting> capabilities = new ArrayList<>();
+        List<BrowserSetting> options = new ArrayList<>();
+        // über den Browser die cap und options holen
+        populateBrowserSpecificSettings(browserName, capabilities, options);
+        // set all capabilities Capabilities
+        populateCapabilities(desiredCapabilities, capabilities);
+        Object browserspecificOptions = null;
+        // set options per browser
+        switch (browserName) {
+            case BrowserType.FIREFOX:
+                FirefoxOptions firefoxOptions = new FirefoxOptions();
+                populateOptions(options, firefoxOptions);
+                browserspecificOptions = firefoxOptions;
+                break;
+            case BrowserType.CHROME:
+                ChromeOptions chromeOptions = new ChromeOptions();
+                populateOptions(options, chromeOptions);
+                browserspecificOptions = chromeOptions;
+                break;
+            default:
+                break;
+        }
+
+        return browserspecificOptions;
+    }
+
+    private void populateOptions(List<BrowserSetting> options, Object browserSpecificOptions) {
+        if (browserSpecificOptions instanceof FirefoxOptions) {
```

Comment:
this called from a switch case, where the browser is known, so why have an if statement with instanceof?